File: brownfield-backend/src/test/java/com/petclinic/vet/VetServiceTest.java
package com.petclinic.vet;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest
@Transactional
class VetServiceTest {

    @Autowired
    private VetService vetService;

    @Test
    void shouldFindAllVets() {
        // when
        List<Vet> vets = vetService.findAll();

        // then
        assertThat(vets).isNotEmpty();
        assertThat(vets.size()).isGreaterThanOrEqualTo(4);
    }

    @Test
    void shouldFindVetById() {
        // given
        List<Vet> allVets = vetService.findAll();
        Vet existingVet = allVets.get(0);

        // when
        Vet foundVet = vetService.findById(existingVet.getId());

        // then
        assertThat(foundVet).isNotNull();
        assertThat(foundVet.getId()).isEqualTo(existingVet.getId());
        assertThat(foundVet.getName()).isEqualTo(existingVet.getName());
        assertThat(foundVet.getSpecialty()).isEqualTo(existingVet.getSpecialty());
    }

    @Test
    void shouldReturnNullWhenVetNotFound() {
        // when
        Vet foundVet = vetService.findById(99999L);

        // then
        assertThat(foundVet).isNull();
    }

    @Test
    void shouldFindVetByName() {
        // when
        Vet foundVet = vetService.findByName("Dr. Sarah Martinez");

        // then
        assertThat(foundVet).isNotNull();
        assertThat(foundVet.getName()).isEqualTo("Dr. Sarah Martinez");
        assertThat(foundVet.getSpecialty()).isEqualTo("Surgery");
    }

    @Test
    void shouldReturnNullWhenVetNameNotFound() {
        // when
        Vet foundVet = vetService.findByName("Dr. NonExistent");

        // then
        assertThat(foundVet).isNull();
    }

    @Test
    void shouldVerifyVetHasCorrectFields() {
        // given
        List<Vet> vets = vetService.findAll();
        assertThat(vets).isNotEmpty();

        // when
        Vet vet = vets.get(0);

        // then
        assertThat(vet.getId()).isNotNull();
        assertThat(vet.getName()).isNotEmpty();
        assertThat(vet.getSpecialty()).isNotEmpty();
    }

    @Test
    void shouldFindVetsWithDifferentSpecialties() {
        // when
        Vet surgeryVet = vetService.findByName("Dr. Sarah Martinez");
        Vet dentistryVet = vetService.findByName("Dr. James Chen");
        Vet generalVet = vetService.findByName("Dr. Emily Rodriguez");
        Vet cardiologyVet = vetService.findByName("Dr. Michael Thompson");

        // then
        assertThat(surgeryVet.getSpecialty()).isEqualTo("Surgery");
        assertThat(dentistryVet.getSpecialty()).isEqualTo("Dentistry");
        assertThat(generalVet.getSpecialty()).isEqualTo("General Practice");
        assertThat(cardiologyVet.getSpecialty()).isEqualTo("Cardiology");
    }
}

File: brownfield-backend/tests/vet/vetService.test.ts
import { describe, it, expect, beforeAll, afterAll } from "vitest";
import { DataSource } from "typeorm";
import { createTestDataSource } from "../testDataSource";
import { VetRepository } from "../../src/vet/vetRepository";
import { VetService } from "../../src/vet/vetService";

let dataSource: DataSource;
let vetService: VetService;

beforeAll(async () => {
  dataSource = await createTestDataSource();
  const vetRepository = new VetRepository(dataSource);
  vetService = new VetService(vetRepository);
});

afterAll(async () => {
  await dataSource.destroy();
});

describe("VetService", () => {
  it("should find all vets", async () => {
    // when
    const vets = await vetService.findAll();

    // then
    expect(vets.length).toBeGreaterThanOrEqual(4);
  });

  it("should find vet by id", async () => {
    // given
    const id = 1;

    // when
    const vet = await vetService.findById(id);

    // then
    expect(vet).not.toBeNull();
    expect(vet?.name).toBe("Dr. Sarah Martinez");
    expect(vet?.specialty).toBe("Surgery");
  });

  it("should return null when vet id does not exist", async () => {
    // given
    const id = 99999;

    // when
    const vet = await vetService.findById(id);

    // then
    expect(vet).toBeNull();
  });

  it("should find vet by name", async () => {
    // when
    const vet = await vetService.findByName("Dr. Sarah Martinez");

    // then
    expect(vet).not.toBeNull();
    expect(vet?.specialty).toBe("Surgery");
  });

  it("should return null when vet name does not exist", async () => {
    // when
    const vet = await vetService.findByName("Dr. NonExistent");

    // then
    expect(vet).toBeNull();
  });

  it("should find vets with different specialties", async () => {
    // when
    const surgeryVet = await vetService.findByName("Dr. Sarah Martinez");
    const dentistryVet = await vetService.findByName("Dr. James Chen");
    const generalVet = await vetService.findByName("Dr. Emily Rodriguez");
    const cardiologyVet = await vetService.findByName("Dr. Michael Thompson");

    // then
    expect(surgeryVet?.specialty).toBe("Surgery");
    expect(dentistryVet?.specialty).toBe("Dentistry");
    expect(generalVet?.specialty).toBe("General Practice");
    expect(cardiologyVet?.specialty).toBe("Cardiology");
  });
});
